db/helpers/generatorUtils.js

```diff
@@ -0,0 +1,142 @@
+  crypted += cipher.final('hex');
+  return crypted;
+} // encrypt
+
+/**
+ * Decrypt encrypted data using passed in secretKey and algorithm
+ * @param  {String} data -  Data to be decrypted
+ * @param  {String} secretKey - Secret key that was used from encryption
+ * @param  {String} algorithm - Name of the encryption algorithm
+ * @returns {String} - encrypted data
+ */
+function decrypt(data, secretKey, algorithm) {
+  const decipher = crypto.createDecipher(algorithm, secretKey);
+  let decrypted = decipher.update(data, 'hex', 'utf-8');
+  decrypted += decipher.final('utf-8');
+  return decrypted;
+} // decrypt
+
+/**
+ * This function fetches the sample generator secret key and the sample
+ * generator encryption algorithm from the glocalConfig model
+ * and returns an object containing the secretKey and algorithm
+ * @param {Model} globalConfigModel - The sequelize Global Config Model
+ * @returns {Object} - containing the secret key and algorithm
+ */
+function getSGEncryptionKeyAndAlgorithm(globalConfigModel) {
+  const sgObj = {};
+  return globalConfigModel.findOne({
+    where: {
+      key: dbConstants.SGEncryptionKey,
+    },
+  })
+  .then((gc) => {
+    if (!gc) {
+      return Promise.resolve(gc);
+    }
+
+    sgObj.secretKey = gc.value;
+    return globalConfigModel.findOne({
+      where: {
+        key: dbConstants.SGEncryptionAlgorithm,
+      },
+    });
+  })
+  .then((gc) => {
+    sgObj.algorithm = gc && gc.value ? gc.value : null;
+    return sgObj;
+  });
+} // getSGEncryptionKeyAndAlgorithm
+
+/**
+ * Encrypts the sample generator context values, if the corresponding
+ * sample generator template context definition value requires encryption.
+ * @param  {GlobalConfig} globalConfigModel - The global config model
+ * @param  {Object} sg - The sample generator object
+ * @param  {Object} sgt - The sample generatot template object.
+ * @returns {Objec} - sample generator object with encrypted context values
+ */
+function encryptSGContextValues(globalConfigModel, sg, sgt) {
+  if (!sg.context || !sgt.contextDefinition) {
+    return Promise.resolve(sg);
+  }
+
+  return getSGEncryptionKeyAndAlgorithm(globalConfigModel)
+  .then((config) => {
+    if (!config || !config.secretKey || !config.algorithm) {
+      return sg;
+    }
+
+    Object.keys(sgt.contextDefinition).forEach((key) => {
+      if (sgt.contextDefinition[key].encrypted && sg.context[key]) {
+        sg.context[key] = encrypt(sg.context[key],
+          config.secretKey, config.algorithm);
+      }
+    });
+
+    return sg;
+  });
+} // encryptSGContextValues
+
+/**
+ * Decrypts the encrypted sample generator context values.
+ * @param  {GlobalConfig} globalConfigModel - The global config model
+ * @param  {Object} sg - The sample generator object
+ * @param  {Object} sgt - The sample generatot template object.
+ * @returns {Objec} - sample generator object with decrypted context values
+ */
+function decryptSGContextValues(globalConfigModel, sg, sgt) {
```

Comment:
Can we move this function in utils/generatorUtils because I would need this function in api/v1/controllers/collectors.js.

Reply:
Moving just this might not really help. I was trying to make this as general purpose as possible. How many of the function exported in the module will you be using?

Reply:
I would need decryptSGContextValues and encrypt functions. I can just import these then. What do you think?

Reply:
What about moving all of under utils and calling them encryptUtils or encryptDecryptUtils?

Reply:
Sounds good to me.